modules/phylowgs/1.0/src/process_phyloWGS_outputs.R

```diff
@@ -0,0 +1,536 @@
+  xlab("Sample") + ylab("VAF") +
+  guides(colour = guide_legend(override.aes = list(alpha = 3)))+
+  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
+        panel.background = element_blank(), axis.line = element_line(colour = "black"),
+        legend.key = element_rect(fill = NA, colour = NA, size = 2),axis.text=element_text(size=16),axis.title=element_text(size=18),
+        plot.margin = margin(1, 1., 1, 1.5, "cm"),axis.title.y = element_text(margin = margin(t = 70, r = 20, b = 50, l = 10)))
+
+print(plotD)
+dev.off() 
+}  
+
+
+plot_vaf_coding(ss,VAF_coding_plot)
+
+#############################################
+#####       Draw the best tree        #######
+#############################################
+
+
+tree_structure_long <- tree_structure %>% 
+  pivot_longer(everything(), 
+               names_to = "parent", 
+               values_to = "node") %>% 
+  mutate(parent = str_remove_all(parent, ".*[.]")) %>% 
+  distinct()  
+
+
+positions_x <- function(parents){
+  x <- 1:length(unique(parents))
+  names(x) <- unique(parents)
+  col_vals <- unname(x[parents])
+  return(col_vals)
+}
+
+tree_structure_long$x <- positions_x(tree_structure_long$parent)
+
+positions_y <- function(tree_df){
+  y = c("0" = 0.5)
+  for(parent in unique(tree_df$parent)){
+    # parent = "1"
+    child_index = 1
+    num_children <- nrow(tree_df[tree_df$parent == parent,])
+    if(num_children == 1){
+      child <- tree_df[tree_df$parent == parent,]$node
+      child_y <- unname(y[parent])
+      names(child_y) <- child
+      y = c(y, child_y)
+      
+    } else {
+      children <- tree_df[tree_df$parent == parent,]$node
+      y_max <- unname(y[parent]) + (0.25 / child_index)
+      y_min <- unname(y[parent]) - (0.25 / child_index)
+      y_range <- seq(y_min, y_max, length.out = length(children))
+      names(y_range) <- children
+      y = c(y, y_range)
+    }
+    child_index = child_index + 1
+  }
+  return(y)
+}
+
+tree_structure_long$y <- unname(positions_y(tree_structure_long)[as.character(tree_structure_long$node)])
+
+tree_structure_long <- add_row(tree_structure_long, parent = "0", node = 0, x = 0, y = 0.5) 
+
+get_ssms <- function(tree_df, best_focal, best_tree_fileID){
+  data <- best_focal[[str_remove_all(best_tree_fileID, "[.].*")]]$populations
+  ssm_vec <- c()
+  for(node in tree_df$node){
+    # node = "1"
+    num_ssms <- data[[as.character(node)]]$num_ssms
+    names(num_ssms) <- as.character(node)
+    ssm_vec <- c(ssm_vec, num_ssms)
+  }
+  return(ssm_vec)
+}
+
+tree_structure_long$num_ssms <- get_ssms(tree_structure_long, best_focal, best_tree_fileID)[as.character(tree_structure_long$node)]
+
+tree_structure_long <- tree_structure_long %>% 
+  mutate(parent = as.numeric(parent)) %>%
+  left_join(select(tree_structure_long, node, xstart = x, ystart = y), 
+            by = c("parent" = "node"))
+
+
+
+ggplot(tree_structure_long, 
```

Comment:
How does this run on the command line without X-forwarding? Not sure if this errors out because it is not saved to a variable here and tries to open graphic device, or it is just goes through? I think `ggsave` below needs plot to be displayed before it can be saved

Reply:
Imlemented.